python/ccxt/base/exchange.py

```diff
@@ -767,12 +767,7 @@ def deep_extend(*args):
 
     @staticmethod
     def filter_by(array, key, value=None):
-        if value:
-            grouped = Exchange.group_by(array, key)
-            if value in grouped:
-                return grouped[value]
-            return []
-        return array
+        return list(filter(lambda x: x[key] == value, array))
```

Comment:
@brandsimon the issue with it is that it lost the ability to filter dicts now: https://github.com/ccxt/ccxt/issues/6150 ← this is obviously a problem with the `.orders` cache (which is `id`-indexed), the EXMO implementation (and a few other) will apply `filter_by` to the `.orders` cache...

Reply:
```
>>> list(filter(lambda x: x['a'] == 1, {'a':1, 'b': 2}))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <lambda>
TypeError: string indices must be integers
```

Reply:
Will need to add `array = Exchange.to_array (array)` to python and php `filter_by`.

Reply:
@kroitor Yes, sorry, my bad. Thanks for notifying me and fixing, I hope I will remember it the next time I edit code like this.

Reply:
@brandsimon it's not your fault, we just lack the tests for the python basecode... Will do our best to add them.